client/src/components/MapElement/MapElement.js

```diff
@@ -168,7 +174,7 @@ export function MapElement({ className }) {
 
     setMap(map)
     return () => map.remove()
-  }, [])
+  }, [isLightMode])
```

Comment:
while trying to repoduce your error without reset view/map/stages, here's some findings:
- error occurred randomly once in 4-16 times
- the markers are good, seems sometimes the map is not getting initialized/setMap is null?
- issue only occurred with vancouver (leading me to suspect the map initialization)
- not sure if this is anything but every time i got the error, it was while I turned light mode on, I notice theres a dependancy added here

we could keep your workaround, it seems tough to reproduce and determine exact cause at this point unless someone is able to spot the issue